analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.vb

```diff
@@ -0,0 +1,503 @@
+﻿Imports System
+Imports System.Collections.Generic
+Imports System.Linq
+
+Class Basics
+    Private Sub NullAssignment()
+        Dim i As Integer? = Nothing
+
+        If i.HasValue Then
+            Console.WriteLine(i.Value)
+        End If
+
+        Console.WriteLine(i.Value)  ' Noncompliant {{'i' is Nothing on at least one execution path.}}
+        '                 ^
+    End Sub
+    Private Sub DereferenceOnValueResult(ByVal i As Integer?)
+        Dim x = i.Value.ToString()  ' Compliant, unknown
+        i = Nothing
+        x = i.Value.ToString()      ' Noncompliant
+    End Sub
+
+    Private Sub NonEmpty()
+        Dim i As Integer? = 42
+
+        If i.HasValue Then
+            Console.WriteLine(i.Value)
+        End If
+
+        Console.WriteLine(i.Value)  ' Compliant
+    End Sub
+
+    Private Sub EmptyConstructor()
+        Dim i As Integer? = New Nullable(Of Integer)()
+
+        If i.HasValue Then
+            Console.WriteLine(i.Value)
+        End If
+
+        Console.WriteLine(i.Value)  ' Noncompliant
+    End Sub
+
+    Private Sub NonEmptyConstructor()
+        Dim i As Integer? = New Nullable(Of Integer)(42)
+
+        If i.HasValue Then
+            Console.WriteLine(i.Value)
+        End If
+
+        Console.WriteLine(i.Value)
+    End Sub
+
+    Private Sub Assignment1(ByVal i1 As Integer?)
+        Dim i2, i3 As Integer?
+
+        i2 = Nothing
+        If i1 = i2 Then
+            Dim x = i1.Value    ' Noncompliant
+        End If
+
+        i1 = 42
+        If i1 <> i2 Then        ' Compliant
+            Dim x = i1.Value
+        End If
+
+        If i1 <> i2 Then
+            Dim x = i2.Value    ' Noncompliant
+        End If
+
+        i3 = Nothing
+        i2 = i3
+        If i1 <> i2 Then
+            Dim x = i2.Value    ' Noncompliant
+        End If
+
+        Dim b1 As Boolean? = Nothing
+
+        If b1 Then              ' Compliant
+        End If
+
+        If Not b1 Then          ' Compliant
+        End If
+    End Sub
+
+    Private Sub AssignmentAndNullComparison(ByVal o As Object)
```

Comment:
Curious about what's your opinion on the topic. 

I still see both conventions (i.e. explicit and implicit `ByVal`) being used in your code, and it's not very clear for me whether you make `ByVal` explicit to mark its use (because it is somehow important for that method parameter to be passed by value) or it is just an inconsistency.

Reply:
It's just an inconsistency. The Telerik converter puts it explicitly and also adds explicit access modifiers everywhere. I try to remove them but overlook some of them. Please point them out (or fix them) if you find any.
I agree with your approach of not marking ByVal explicitly and keeping access modifiers to a minimum to declutter the code.